lib/gems/pending/VMwareWebService/MiqVimDataStore.rb

```diff
@@ -287,7 +286,14 @@ def get_file_content(filepath)
     /(\d*)\.(\d*)/ =~ @invObj.about['version']
     raise "get_file_content not supported on [#{@invObj.about['fullName']}]" if $1.to_i < 3 || ($1.to_i == 3 && $2.to_i < 5)
 
-    fileUrl = "https://#{@invObj.server}/folder/#{MIQEncode.base64Encode(filepath)}?dsName=#{MIQEncode.base64Encode(@name)}"
+    uri_parts = {
+      :scheme => "https",
+      :host   => @invObj.server,
+      :path   => "/folder/#{filepath}",
+      :query  => "dsName=#{@name}"
+    }
+
+    fileUrl = URI::Generic.build(uri_parts).to_s
```

Comment:
> Also, do you know what special characters can be in the filepath?

It is a VM name so some of the special characters are already URI escaped, but you're right it is definitely safer to `URI.encode()` first